### PR TITLE
Role Authentication Middleware

### PR DESCRIPTION
**Pull request by justinfineberg**
Functionality: We set up a middleware that ensures the user's who hit the endpoints actually have the proper authorization to hit this endpoint. It's a simple middleware that receives the arguments of the user types that should be allowed access to the endpoint. 
For instance, an endpoint that should only have 'instructor' access, the middleware can be called in the router by writing "roleAuthentication('instructor')". 

https://www.loom.com/share/0499a0521158402bbafc51fb23b98f5f

### CODE REVIEW

File: api/middleware/roleAuthentication.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const roleAuthentication = (...args) => (req, res, next) => {
+  // role type is inside req.profile.body
+  //const { type } = req.profile
```
Comment (AshleyCoder3) [inline api/middleware/roleAuthentication.js]: may want to remove duplicate line

**Review by AshleyCoder3 — CHANGES_REQUESTED** [review]
Looks good after that one-line fix. line 3

File: api/middleware/roleAuthentication.js (hunk 1 above)
Comment (ehsan2793) [reply api/middleware/roleAuthentication.js]: yeah agree with Ashley you don't want to have those commented out code
I would say drop this pr fix the issue and send another PR
but overall great job Justin

**Review by ehsan2793 — APPROVED** [review]
awesome job, thanks for removing those comments.
cant wait to test it out

**Review by AshleyCoder3 — APPROVED** [review]
Looks great

**Pull request merged**